Bug: `test_k8s_disjoin_join.py` doesn't teardown iptables to original state

### Is it platform specific

generic

### Importance or Severity

Medium

### Description of the bug

When `kubesonic/test_k8s_join_disjoin.py` is run, there are some leftover config in `iptables` and `ip6tables`.

This could cause errors in `cacl/test_cacl_application.py` as it checks `iptables -S` and `ip6tables -S` for expected results (or similar tests that may use `iptables`/`ip6tables`).

Good to note `config reload` does reset the iptables, but is not guaranteed to run (and shouldn't be run just to teardown)

### Steps to Reproduce

On a clean DUT (config reloaded) run `iptables -S; ip6tables -S` OR `iptables-save; ip6tables-save`
Run `kubesonic/test_k8s_disjoin_join.py` test
run `iptables -S; ip6tables -S` OR `iptables-save; ip6tables-save` again
See difference

To see it causing test failures, run `kubesonic/test_k8s_disjoin_join.py` and `cacl/test_cacl_application.py` consecutively

### Actual Behavior and Expected Behavior

`kubesonic/test_k8s_disjoin_join.py` should teardown to leave iptables as it was originally

### Relevant log output

```shell
`cacl/test_cacl_application.py` failure when running it after kubesonic test:

E       Failed: Unexpected iptables rules: {'-A KUBE-FIREWALL -m comment --comment "kubernetes firewall for dropping marked packets" -m mark --mark 0x8000/0x8000 -j DROP', '-A INPUT -j KUBE-FIREWALL', '-N KUBE-FIREWALL', '-N KUBE-KUBELET-CANARY', '-A KUBE-FIREWALL ! -s 127.0.0.0/8 -d 127.0.0.0/8 -m comment --comment "block incoming localnet connections" -m conntrack ! --ctstate RELATED,ESTABLISHED,DNAT -j DROP', '-A OUTPUT -j KUBE-FIREWALL'}
```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/kubesonic/test_k8s_join_disjoin.py
+++ b/tests/kubesonic/test_k8s_join_disjoin.py
@@ -277,6 +277,10 @@ def clean_configdb_k8s_table(duthost):
 
 @pytest.fixture()
 def setup_and_teardown(duthost, vmhost, creds):
+    # Capture initial iptables
+    duthost.shell("sudo iptables-save > /tmp/iptables_rules_before")
+    duthost.shell("sudo ip6tables-save > /tmp/ip6tables_rules_before")
+
     check_image_type_supported(duthost)
     check_dut_k8s_version_supported(duthost)
     logger.info("Start to setup test environment")
@@ -345,6 +349,12 @@ def setup_and_teardown(duthost, vmhost, creds):
         # Clean up the current env
         remove_k8s_master(vmhost)
         remove_minikube(vmhost)
+
+    # Restore original iptables and cleanup
+    duthost.shell("sudo iptables-restore < /tmp/iptables_rules_before")
+    duthost.shell("sudo ip6tables-restore < /tmp/ip6tables_rules_before")
+    duthost.shell("rm -f /tmp/iptables_rules_before")
+    duthost.shell("rm -f /tmp/ip6tables_rules_before")
 
 
 def trigger_join_and_check(duthost, vmhost):

